apps/vmq_server/src/vmq_queue.erl

```diff
@@ -815,9 +815,10 @@ send_notification(#session{pid=Pid} = Session) ->
     vmq_mqtt_fsm:send(Pid, {mail, self(), new_data}),
     Session#session{status=passive}.
 
-cleanup_session(SubscriberId, #session{queue=#queue{queue=Q}}) ->
+cleanup_session(SubscriberId, #session{queue=#queue{queue=Q, backup=BQ}}) ->
     _ = vmq_metrics:incr_queue_unhandled(queue:len(Q)),
```

Comment:
Good question. I'd say we shouldn't as it would be guesswork on the part of the queue and I suppose those messages would only be unhandled if the fsm crashes and doesn't do the right thing?